### PR TITLE
cmake: make install locations configurable

### LINKED ISSUE #3589: include could not find load file: HPX_Utils.cmake
On Fedora 29:
```bash
dnf install hpx-openmpi-devel
module load mpi/openmpi-x86_64
```

Create a `CMakeLists.txt` with:
```cmake
cmake_minimum_required(3.1)
find_package(HPX)
```
now run:
```bash
cmake .
```

one gets:
``` 
-- The C compiler identification is GNU 8.2.1
-- The CXX compiler identification is GNU 8.2.1
-- Check for working C compiler: /usr/lib64/ccache/cc
-- Check for working C compiler: /usr/lib64/ccache/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/lib64/ccache/c++
-- Check for working CXX compiler: /usr/lib64/ccache/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  cmake_minimum_required called with unknown argument "3.1".


CMake Error at /usr/lib64/openmpi/lib/cmake/HPX/HPXMacros.cmake:8 (include):
  include could not find load file:

    HPX_Utils.cmake
Call Stack (most recent call first):
  /usr/lib64/openmpi/lib/cmake/HPX/HPXConfig.cmake:13 (include)
  CMakeLists.txt:2 (find_package)


CMake Error at /usr/lib64/openmpi/lib/cmake/HPX/HPXMacros.cmake:70 (include):
  include could not find load file:

    HPX_SetupAllocator
Call Stack (most recent call first):
  /usr/lib64/openmpi/lib/cmake/HPX/HPXConfig.cmake:73 (hpx_setup_allocator)
  CMakeLists.txt:2 (find_package)


-- Configuring incomplete, errors occurred!
See also "/tmp/hpx/CMakeFiles/CMakeOutput.log".
```

Looking at `/usr/lib64/openmpi/lib/cmake/HPX/HPXMacros.cmake` it seems the hard-coded `CMAKE_MODULE_PATH` in there is wrong. It is set to `/usr/lib/cmake/HPX` instead of `/usr/lib64/openmpi/lib/cmake/HPX`.

### PR DESCRIPTION
**Pull request by junghans**
## Proposed Changes

Use `GNUInstallDirs` to make install location more configurable.

## Any background context you want to provide?

This would be helpful for Fedora packaging in the future.

Fixed #3589

### CODE REVIEW

File: cmake/HPX_AddExecutable.cmake [hunk 1]
```diff
@@ -149,7 +149,7 @@ function(add_hpx_executable name)
     endif()
   endif()
 
-  set_target_properties(${name}_exe PROPERTIES OUTPUT_NAME ${name})
+  set_target_properties(${name}_exe PROPERTIES OUTPUT_NAME "${HPX_OUTPUT_PREFIX}${name}")
```
Comment (msimberg) [inline cmake/HPX_AddExecutable.cmake]: Is this meant for prefixing e.g. examples with `hpx_` when installing them systemwide? And meant to be set by the user? Would you mind adding it as an option somewhere around [here](https://github.com/STEllAR-GROUP/hpx/blob/c1478e81321c67f55ca029640b75582c1cb70231/CMakeLists.txt#L216-L248)?

**Comment by msimberg** [comment]
@hkaiser do you know if Windows/VS is going to do the right thing with this?

**Comment by hkaiser** [comment]
@msimberg that's on my list to check. I would assume the cmake guys have done 'the right thing', but who knows...

File: cmake/HPX_AddExecutable.cmake (hunk 1 above)
Comment (junghans) [reply cmake/HPX_AddExecutable.cmake]: Done.

Comment (msimberg) [reply cmake/HPX_AddExecutable.cmake]: Thanks! Could you actually make it `HPX_WITH_OUTPUT_PREFIX` (Is `HPX_WITH_EXECUTABLE_PREFIX` clearer? That's what you write in the description.) instead to be consistent with other options. Sorry, should've made that clear the first time around!

Comment (junghans) [reply cmake/HPX_AddExecutable.cmake]: Done.

Comment (msimberg) [reply cmake/HPX_AddExecutable.cmake]: Perfect, thanks!

**Comment by junghans** [comment]
@msimberg do you know why the CI is failing?

**Comment by msimberg** [comment]
The `migrate_component` test has been problematic already for some time so it's unlikely that the error is caused by this PR. I've just restarted the build.

@sithhell usually `migrate_component` times out, but this time an assertion fails (https://circleci.com/gh/STEllAR-GROUP/hpx/50845). Have you seen that before?

@hkaiser afaict from the documentation this will do the same thing on Windows and elsewhere. It just uses (GNU) standard directories *under* `CMAKE_INSTALL_PREFIX`, but the default value of `CMAKE_INSTALL_PREFIX` isn't changed by this. So unless the old directory names were chosen to be typical directory names for Windows I think this is a step up also on Windows.

**Pull request merged**